derive_client/bridge/enums.py

```diff
@@ -0,0 +1,57 @@
+"""Enums used in the bridge module."""
+
+from enum import IntEnum, StrEnum, auto
+
+
+class TxStatus(IntEnum):
+    FAILED = 0
+    SUCCESS = 1
+
+
+class ChainID(IntEnum):
+    ETH = 1
+    OPTIMISM = 10
+    LYRA = 957
+    BASE = 8453
+    MODE = 34443
+    ARBITRUM = 42161
+    BLAST = 81457
+
+    @classmethod
+    def _missing_(cls, value):
+        try:
+            int_value = int(value)
+            return next(member for member in cls if member == int_value)
+        except (ValueError, TypeError, StopIteration):
+            return super()._missing_(value)
+
+
+class Currency(StrEnum):
+    @staticmethod
+    def _generate_next_value_(name: str, start: int, count: int, last_values: list[str]):
```

Comment:
I had to google what this does;

Not sure why again this is better than a normal enum would really love to understand

Reply:
yeah this allows me to use `auto()`. I want the `StrEnum` values here to be identical to the names; case sensitive. This is particularly useful if you have custom Enums with more complex member types, or when using generics

Reply:
@8ball030 regarding [this comment](https://github.com/8ball030/derive_client/pull/11#discussion_r2041869956) in #11 , These are not the same
- [`Currency`](https://github.com/8ball030/derive_client/blob/ccd5b87a38ccde44fbe9f15f6177ffa2fb231421/derive_client/bridge/enums.py#L29-L48) is for token products (weETH, sUSDe, etc.)
- [`UnderlyingCurrency`](https://github.com/8ball030/derive_client/blob/ccd5b87a38ccde44fbe9f15f6177ffa2fb231421/derive_client/enums.py#L16-L29) is for base tokens (ETH, BTC, etc.)